test/runtests.jl

```diff
@@ -15,1923 +15,1699 @@ end
+    @testset for elty in [Float32, Float64, Complex64, Complex128]
+        A = rand(elty, m)
+        @test ndims(A) == 1
+        n1 = length(A)
+        d_A = CudaArray(A)
+        cuda_asum1 = CUBLAS.asum(n1,d_A,1)
+        cuda_asum2 = CUBLAS.asum(d_A)
+        host_asum = sum(abs(real(A)) + abs(imag(A)))
+        @test host_asum ≈ cuda_asum1
+        @test host_asum ≈ cuda_asum2
+    end
+end
+
+@testset "axpy!" begin
+    # test axpy!
+    function test_axpy!_1(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        CUBLAS.axpy!(n1,alpha,d_A,1,d_B1,1)
+        B1 = to_host(d_B1)
+        host_axpy = alpha*A + B
+        @test host_axpy ≈ B1
+    end
+    test_axpy!_1(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_1(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_1(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_1(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))
+
+    function test_axpy!_2(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        CUBLAS.axpy!(alpha,d_A,d_B1)
+        B1 = to_host(d_B1)
+        host_axpy = alpha*A + B
+        @test host_axpy ≈ B1
+    end
+    test_axpy!_2(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_2(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_2(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_2(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))
+
+    #=function test_axpy!_3(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        CUBLAS.axpy!(alpha,d_A,1:2:n1,d_B1,1:2:n1)
+        B1 = to_host(d_B1)
+        host_axpy = B
+        host_axpy[1:2:n1] = alpha*A[1:2:n1] + B[1:2:n1]
+        @test_approx_eq(host_axpy,B1)
+    end
+    test_axpy!_3(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_3(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_3(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_3(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))
+
+    function test_axpy!_4(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        r = 1:div(n1,2)
+        CUBLAS.axpy!(alpha,d_A,r,d_B1,r)
+        B1 = to_host(d_B1)
+        host_axpy = B
+        host_axpy[r] = alpha*A[r] + B[r]
+        @test_approx_eq(host_axpy,B1)
+    end
+    test_axpy!_4(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_4(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_4(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_4(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))=#
+end
+
+@testset "iamax and iamin" begin
+    @testset for elty in [Float32, Float64, Complex64, Complex128]
+        A = rand(elty, m)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        Aabs = blasabs(A)
```

Comment:
That's a user-defined function at the top of the file, but it needs an `abs.` inside :)